src/ServiceInsight/Framework/Settings/ApplicationConfiguration.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace ServiceInsight.Framework.Settings
+{
+    using System;
+    using System.Configuration;
+
+    public class ApplicationConfiguration
+    {
+        public static bool SkipCertificateValidation => Convert.ToBoolean(ConfigurationManager.AppSettings["SkipCertificateValidation"]);
```

Comment:
If the string is not a valid boolean representation, we'll get a `System.FormatException` with the message "String was not recognized as a valid Boolean.". While the stack trace could indicate what the problem is, I guess this could be subject to optimisation and may not be guaranteed to do so. I think it would be better to catch any conversion exception and wrap it in an exception which calls out the name of the configuration item.

Reply:
I have changed the way the values are parsed. Now you get a better exception details/message. The exception will be thrown if the value can not be parsed or the key is missin]g altogether, which I find to cause fewer surprises.